app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt

```diff
@@ -18,6 +31,44 @@ class CustomPrescriptionEntryEffectHandler @AssistedInject constructor(
       : ObservableTransformer<CustomPrescriptionEntryEffect, CustomPrescriptionEntryEvent> {
     return RxMobius
         .subtypeEffectHandler<CustomPrescriptionEntryEffect, CustomPrescriptionEntryEvent>()
+        .addTransformer(SaveCustomPrescription::class.java, saveCustomPrescription(schedulersProvider.io()))
         .build()
   }
+
+  private fun saveCustomPrescription(io: Scheduler): ObservableTransformer<SaveCustomPrescription, CustomPrescriptionEntryEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(io)
+          .map {
+            val user = userSession.loggedInUserImmediate()!!
+            val currentFacility = facilityRepository.currentFacilityImmediate(user)!!
+            SavePrescription(
+                patientUuid = it.patientUuid,
+                name = it.drugName,
+                dosage = it.dosage,
+                facility = currentFacility
+            )
```

Comment:
This has been refactored in the later commits to extract creating of the `SavePrescription` object.
Does it make sense to not do this anymore?

Reply:
Yeah, I was questioning the need for the `SavePrescription` object itself since it doesn't actually anything. The facility can be queried in line when making the `savePrescription` call.

Reply:
Oh I got what you mean. I will make this change.